features/docs/stap-definities.feature

```diff
@@ -0,0 +1,326 @@
+      | naam                  | waarde |
+      | geslachtsnaam (02.40) | Jansen |
+      Dan zijn de gegenereerde SQL statements
+      | key          | text                                                                                                                                     | values                 |
+      | inschrijving | INSERT INTO public.lo3_pl(pl_id,mutatie_dt,geheim_ind) VALUES((SELECT MAX(pl_id)+1 FROM public.lo3_pl),current_timestamp,$1) RETURNING * | 0                      |
+      | persoon      | INSERT INTO public.lo3_pl_persoon(stapel_nr,volg_nr,persoon_type,burger_service_nr,geslachts_naam) VALUES($1,$2,$3,$4,$5)                | 0,0,P,000000012,Jansen |
+
+  Rule: En de persoon is gewijzigd naar de volgende gegevens
+
+    Scenario: Gewijzigd persoon
+      Gegeven de persoon met burgerservicenummer '000000012' heeft de volgende gegevens
+      | naam                  | waarde |
+      | geslachtsnaam (02.40) | Jansen |
+      En de persoon is gewijzigd naar de volgende gegevens
+      | naam                  | waarde    |
+      | geslachtsnaam (02.40) | Pietersen |
+      Dan zijn de gegenereerde SQL statements
+      | key          | text                                                                                                                                     | values                 |
+      | inschrijving | INSERT INTO public.lo3_pl(pl_id,mutatie_dt,geheim_ind) VALUES((SELECT MAX(pl_id)+1 FROM public.lo3_pl),current_timestamp,$1) RETURNING * | 0                      |
+      | persoon      | INSERT INTO public.lo3_pl_persoon(stapel_nr,volg_nr,persoon_type,burger_service_nr,geslachts_naam) VALUES($1,$2,$3,$4,$5)                | 0,1,P,000000012,Jansen |
+      |              | INSERT INTO public.lo3_pl_persoon(stapel_nr,volg_nr,persoon_type,geslachts_naam) VALUES($1,$2,$3,$4)                                     | 0,0,P,Pietersen        |
+
+  Rule: En de persoon is gecorrigeerd naar de volgende gegevens
+
+    Scenario: Gecorrigeerd persoon
+      Gegeven de persoon met burgerservicenummer '000000012' heeft de volgende gegevens
+      | naam                  | waarde |
+      | geslachtsnaam (02.40) | Jansen |
+      En de persoon is gecorrigeerd naar de volgende gegevens
+      | naam                  | waarde  |
+      | geslachtsnaam (02.40) | Janssen |
+      Dan zijn de gegenereerde SQL statements
+      | key          | text                                                                                                                                     | values                   |
+      | inschrijving | INSERT INTO public.lo3_pl(pl_id,mutatie_dt,geheim_ind) VALUES((SELECT MAX(pl_id)+1 FROM public.lo3_pl),current_timestamp,$1) RETURNING * | 0                        |
+      | persoon      | INSERT INTO public.lo3_pl_persoon(stapel_nr,volg_nr,persoon_type,burger_service_nr,geslachts_naam,onjuist_ind) VALUES($1,$2,$3,$4,$5,$6) | 0,1,P,000000012,Jansen,O |
```

Comment:
Lees ik het goed dat je eerst de statements samenstelt op van alle Gegeven stappen bij elkaar?

Reply:
Dat klopt. Dat is iets wat jij wou, toch? De code is complexer, maar dat scheelt in aantal sql statements die worden uitgevoerd en dat is volgens mij ook iets wat je destijds opmerkte mbt performance van een test